### DISCUSSION TITLE
[v5] is this bug in before fetch paginate?

### DISCUSSION THREAD

**Discussion by sasha172017** [adonisjs/core#1994] [General]
if i write   

@beforeFetch()
  static get(query) {
    query
      .orWhereHas('translations', query => {
        query.where('name', 'ILiKE', '%%')
      })
      .orWhereHas('translations', query => {
        query.where('description', 'ILiKE', '%%')
      })
  }

in paginate total count will be without whereHas.... what will do it?

**Comment by Obapelumi** [comment]
Saw this issue too when using the beforeFetch hook. Also, it seems to add the query here before all the other queries. So if you have an orWhere clause this hook doesn't work.
